handshake_messages.go

```diff
@@ -34,31 +34,32 @@ type clientHelloMsg struct {
 	psks                         []psk
 	pskKeyExchangeModes          []uint8
 	earlyData                    bool
+	delegatedCredential          bool
 }
 
 // Helpers
 
 // Marshalling of signature_algorithms extension see https://tools.ietf.org/html/rfc5246#section-7.4.1.4.1
 // for more details. Extension is serialized in data buffer
 // Function advances data slice and returns it, so that it can be used for further processing
-func marshallExtensionSignatureAlgorithms(data []byte, sigSchemes []SignatureScheme) ([]byte) {
-    data[0] = byte(extensionSignatureAlgorithms >> 8)
-    data[1] = byte(extensionSignatureAlgorithms)
-    l := 2 + 2*len(sigSchemes)
-    data[2] = byte(l >> 8)
-    data[3] = byte(l)
-    data = data[4:]
-
-    l -= 2
-    data[0] = byte(l >> 8)
-    data[1] = byte(l)
-    data = data[2:]
-    for _, sigAlgo := range sigSchemes {
-        data[0] = byte(sigAlgo >> 8)
-        data[1] = byte(sigAlgo)
-        data = data[2:]
-    }
-    return data
+func marshallExtensionSignatureAlgorithms(data []byte, sigSchemes []SignatureScheme) []byte {
+	data[0] = byte(extensionSignatureAlgorithms >> 8)
+	data[1] = byte(extensionSignatureAlgorithms)
+	l := 2 + 2*len(sigSchemes)
+	data[2] = byte(l >> 8)
+	data[3] = byte(l)
+	data = data[4:]
+
+	l -= 2
+	data[0] = byte(l >> 8)
+	data[1] = byte(l)
+	data = data[2:]
+	for _, sigAlgo := range sigSchemes {
+		data[0] = byte(sigAlgo >> 8)
+		data[1] = byte(sigAlgo)
+		data = data[2:]
+	}
+	return data
```

Comment:
😭

Reply:
also, marshal is spelt with one L

Reply:
What exactly is the complaint here? This tab was added by vim's built-in linter, so I assume this change is "idomatic". I'd have to disable my linter to fix this.

Thx for noticing the spelling.

Reply:
I'm glad you fixed it, sad it happened.

Reply:
I did `s/arshall/arshal/g` in `handshake_messages.go`. Note that this spelling error existed before I touched the code. So technically this change makes the PR not minimal

Reply:
Minimal PR is good idea. This should be done in 2 seprated commits:
1. https://github.com/cloudflare/tls-tris/pull/96
2. https://github.com/cloudflare/tls-tris/pull/97

Thanks for noticing

Reply:
Ack. See my comment on PR #96.